part-i-getting-started/quick-start.md

```diff
@@ -0,0 +1,355 @@
+
+```java
+public class RedeemedEvt {
+    
+    private final String id;
+    private final Integer amount;
+    
+    public RedeemedEvt(String id, Integer amount) {
+        this.id = id;
+        this.amount = amount;
+    }
+    
+    public String getId() {
+        return id;
+    }
+        
+    public Integer getAmount() {
+        return amount;
+    }
+}
+```
+
+> **Note** Command/Event definition looks a bit cumbersome, this can be avoided using hierarchy between commands/events, or using libraries like [lombok](https://projectlombok.org/), or using [kotlin](https://kotlinlang.org/) which provides more concise way of defining commands/events.
+
+### Aggregate
+
+Now that we have our commands and events which represent the API of our application, we can create Giftcard aggregate:
+
+```java
+public class GiftCard {
+
+    @AggregateIdentifier // (1)
+    private String id;
+    private int remainingValue;
+
+    public GiftCard() {
+        // (2)
+    }
+	
+	@CommandHandler
+    public GiftCard(IssueCmd cmd) {        
+        if(cmd.getAmount() <= 0) throw new IllegalArgumentException("amount <= 0");
+        AggregateLifecycle.apply(new IssuedEvt(cmd.getId(), cmd.getAmount()));
+    }
+	
+	@EventSourcingHandler
+    public void on(IssuedEvt evt) {
+        id = evt.getId();
+        remainingValue = evt.getAmount();
+    }
+
+    @CommandHandler
+    public void handle(RedeemCmd cmd) {
+        if(cmd.getAmount() <= 0) throw new IllegalArgumentException("amount <= 0");
+        if(cmd.getAmount() > remainingValue) throw new IllegalStateException("amount > remaining value");
+        AggregateLifecycle.apply(new RedeemedEvt(id, cmd.getAmount()));
+    }
+
+    @EventSourcingHandler
+    public void on(RedeemedEvt evt) {
+        remainingValue -= evt.getAmount();
+    }
+}
+```
+
+(1) `@AggregateIdentifier` annotation tells Axon that annotated field will be used as identifier of the Aggregate.
+
+(2) If you are using Axon for Event Sourcing, default constructor is needed so Axon can instantiate the Aggregate and apply all sourced events.
+
+> **Note** All business logic / rules are defined in the `@CommandHandler`s, and all state changes are defined in the `@EventSourcingHandler`s. The reason for this is when we want to get the current state of event-sourced Aggregate, we have to apply all sourced events - we have to invoke `@EventSourcingHandler`s. If the state of our Aggregate is changed outside of `@EventSourcingHandler`s it will not be reflected when we do a replay.
+
+### Query Model
+
+Once we have our Aggregate defined which processes commands and fires events, we can create a Query Model (Projection) based on those fired events. Let's say that we want to build a view that has summary of Giftcards. In order to do that, view will issue a query to our Query Model to retrieve necessary information about Giftcards. We will use good old `List` Java structure as our storage. `CardSummary` class could look like this:
```

Comment:
Small typos in here I think: 
1. `...that we want to build a view that has **a** summary of GiftCards.`
2. `In order to do that, **the** view will...`
3. `**The** CardSummary class could look...`